sdlib/d/gc/emap.d

```diff
@@ -39,7 +38,33 @@ public:
 	}
 
 	void clear(Extent* extent) shared {
-		tree.clearRange(extent.address, extent.size);
+		clear(extent.address, extent.size);
+	}
+
+	bool resize(Extent* extent, size_t size) shared {
+		assert(size > 0, "Invalid size!");
+
+		if (extent.size == size) {
+			return true;
+		}
+
+		if (extent.size > size) {
+			clear(extent.address + size, extent.size - size);
+			return true;
+		}
+
+		auto pd = PageDescriptor(extent, extent.extentClass);
+
+		return map(extent.address + extent.size, size - extent.size, pd);
+	}
```

Comment:
The caller will have to compute this, if for no other reason that the shrink need to happen before the extent is updated and vice versa. So it doesn't look judicious to have a unified function here.